src/Internal/Payment/State/InitialState.php

```diff
@@ -23,34 +32,118 @@ class InitialState extends AbstractPaymentState {
-	 * @param OrderService $order_service Service for order-related actions.
+	 * @param StateFactory                 $state_factory           Factory for payment states.
+	 * @param OrderService                 $order_service           Service for order-related actions.
+	 * @param WC_Payments_Customer_Service $customer_service        Service for managing remote customers.
+	 * @param Level3Service                $level3_service          Service for Level3 Data.
+	 * @param PaymentRequestService        $payment_request_service Connection with the server.
 	 */
 	public function __construct(
 		StateFactory $state_factory,
-		OrderService $order_service
+		OrderService $order_service,
+		WC_Payments_Customer_Service $customer_service,
+		Level3Service $level3_service,
+		PaymentRequestService $payment_request_service
 	) {
 		parent::__construct( $state_factory );
 
-		$this->order_service = $order_service;
+		$this->order_service           = $order_service;
+		$this->customer_service        = $customer_service;
+		$this->level3_service          = $level3_service;
+		$this->payment_request_service = $payment_request_service;
 	}
 
 	/**
 	 * Initialtes the payment process.
 	 *
+	 * @param PaymentRequest $request    The incoming payment processing request.
 	 * @return CompletedState            The next state.
 	 * @throws StateTransitionException  In case the completed state could not be initialized.
 	 * @throws ContainerException        When the dependency container cannot instantiate the state.
 	 * @throws Order_Not_Found_Exception Order could not be found.
+	 * @throws PaymentRequestException   When data is not available or invalid.
 	 */
-	public function process() {
-		$order_id       = $this->get_context()->get_order_id();
-		$payment_method = $this->get_context()->get_payment_method();
-		$this->order_service->set_payment_method_id( $order_id, $payment_method->get_id() );
+	public function process( PaymentRequest $request ) {
+		$context  = $this->get_context();
+		$order_id = $context->get_order_id();
 
+		// Populate basic details from the request.
+		$this->populate_context_from_request( $request );
+
+		// Start by setting up all local objects.
+		$this->order_service->populate_context_from_order( $order_id, $context );
+		$context->set_metadata( $this->order_service->get_payment_metadata( $order_id ) );
+		$context->set_level3_data( $this->level3_service->get_data_from_order( $order_id ) );
+
+		// Customer management involves a remote call.
+		$customer_id = $this->customer_service->get_or_create_customer_id_from_order(
+			$context->get_user_id(),
+			$this->order_service->_deprecated_get_order( $order_id )
+		);
+		$context->set_customer_id( $customer_id );
+
+		// Payments are currently based on intents, request one from the API.
+		try {
+			$intent = $this->payment_request_service->create_intent( $context );
+		} catch ( Invalid_Request_Parameter_Exception | Extend_Request_Exception | Immutable_Parameter_Exception $e ) {
```

Comment:
Yes, now that we've got all the context population in the same place, this would be the place to handle exceptions. We'll also need to discuss (and implement) the separation between types of errors, how the exception is logged (incl. the logger class), and what is displayed to the buyer.

I have this in my list, and there will be a few separate issues related to it.